### ISSUE TITLE
sign-compare warning in Marshal.hpp

### ISSUE THREAD

**Issue by richey-v** [cross-language-cpp/djinni-support-lib#13]
Upgraded from Ubuntu18.04 to Ubuntu20.04 and am now getting a (new) `sign-compare` warning in Marshall.hpp:
```
djinni/jni/Marshal.hpp: In static member function ‘static djinni::LocalRef<_jbyteArray*> djinni::Binary::fromCpp(JNIEnv*, const CppType&)’:
djinni/jni/Marshal.hpp:246:29: warning: comparison of integer expressions of different signedness: ‘std::vector<unsigned char>::size_type’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  246 |             assert(c.size() <= std::numeric_limits<jsize>::max());
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Most likely due to and upgrade of clang.  I'm going to try to fix this in a fork and issue a PR.  I'll also try to update w/ compiler details.

**Comment by richey-v** [comment]
clang-10

**Comment by richey-v** [comment]
personally, I'd like some real error handling, rather than `assert`... the code below that and the other identical `assert`s will overflow buffers!

**Comment by a4z** [comment]
Thanks for the report Will, I agree that warning should not be. 
Will be removed. That's the easy part.

About an alternative error strategy, .... that's way more tricky to get right.
In real the checks are not there (release builds), so they are more type of documentation.
But it's probably a good idea to go over them and verify

**Comment by a4z** [comment]
The warning should be removed after the merge of #15 , but there is still the suggestion to remove the assert statement and replace it with something different.

Would you be ok with creating a new issue @richey-v please, for the assert question , and closing this one since the warning is resolved? 
This would give a nicer change log history also.

**Comment by richey-v** [comment]
Warnings resolved by #15.  #16 opened for the enhancement effort.

@a4z - could you please add the enhancement and any other labels to #16?  I don't see where I can do that.
